designprep/src/main/java/SnakeAndLadderGame.java

```diff
@@ -0,0 +1,131 @@
+import gamemodel.GameEnhancement;
+import gamemodel.GameModel;
+import gamemodel.Player;
+
+import java.util.*;
+
+/**
+ * @author r.shukla
+ */
+public class SnakeAndLadderGame {
+
+    private static final int DICE_FACES = 6;
+    private static Map<Integer, GameModel> snakesMap = new HashMap<>();
+    private static Map<Integer, GameModel> ladderMap = new HashMap<>();
+    private static Queue<Player> playerQueue = new LinkedList<>();
+    private static final int MAX_BOARD_CELL = 100;
+
+    /**
+     * ENUM for game factors
+     */
+    private enum GameFactors { SNAKES, LADDERS }
+
+    /**
+     *
+     * @param args
+     */
+    public static void main(String[] args) {
+        scannGameInfo();
+        playGame();
+    }
+
+    private static void scannGameInfo() {
+        Scanner scanner = new Scanner(System.in);
```

Comment:
getting values from user should not be a part of this class. should be in a main driver